JSON formatted error responses and end to end tests

iloire opened the pull request:
By wrapping errors in JSON responses, it is feasible to add custom data (like error source) to the response, and upstream services can easily determine whenever are the calls to the preview url and not calls to the iframely server (or any other proxy/service in the middle) the ones failing.

This is pretty helpful in iframely cloud, but it is not available in the open source version yet.

Happy to hear suggestions about the response object signature.

Running the e2e tests requires a version of node that supports ES2015 features, though the modifications introduced in the server don't.

iloire commented:
I can change the test to be compatible with 0.12, but I would wait until getting your feedback about the overall approach. Thanks

iparamonau commented:
Thanks @iloire! The error codes responses look good, except for 410, which we generously use for `gone` content - e.g. when a tweet gets removed. Would be easier to accept the PR in two parts.

I will let @nleush take a look into tests part. Looks quite promising too.

iloire commented:
Hi Ivan, thanks for the response.

The end to end tests now run in 0.10 and above (hence passing in Travis CI).
I also addressed your comment and return a generic 'Gone' for 410 responses.

Happy to address any other feedback you may have.

nleush commented:
Thanks @iloire it all looks good. Merged to main develop branch now.
